### PR TITLE
[expat] Add data for release 2.3.0.

### PR DESCRIPTION
**Pull request by datalogics-robb**
Specify library name and version:  **expat/2.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by datalogics-robb** [comment]
Build failures are for VS 2015 - and are due to expat-2.3.0 explicitly dis-allowing it in CMake, as documented here:

  https://github.com/libexpat/libexpat/tree/master/expat

Is there some way we can limit the CI to the compilers expat supports, or do we need to remove this restriction with a patch?

**Comment by SpaceIm** [comment]
raise ConanInvalidConfiguration in `validate()` if version of expat is greater or equal than 2.3.0, compiler is Visual Studio and version of compiler is less than 15.

**Comment by datalogics-robb** [comment]
@SpaceIm I attempted the version check & raise, but that seemed to have issues that weren't obvious to me, so I changed over to a patch that modifies the CMake error to a warning.

File: recipes/expat/all/conanfile.py [hunk 1]
```diff
@@ -34,6 +33,9 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/expat/all/conanfile.py:37]: in `build()`, not in `source()` please

File: recipes/expat/all/conandata.yml [hunk 2]
```diff
@@ -11,3 +11,11 @@ sources:
...
     sha256: bf42d1f52371d23684de36cc6d2f0f1acd02de264d1105bdc17792bbeb7e7ceb
     url: https://github.com/libexpat/libexpat/releases/download/R_2_2_10/expat-2.2.10.tar.gz
+  "2.3.0":
+    sha256: 89df123c62f2c2e2b235692d9fe76def6a9ab03dbe95835345bf412726eb1987
+    url: https://github.com/libexpat/libexpat/releases/download/R_2_3_0/expat-2.3.0.tar.gz
+patches:
+  2.3.0:
+    - patch_file: "patches/relax-vs-restriction.patch"
+      base_path: "source_subfolder"
+      strip: 2
```
Comment (SpaceIm) [inline recipes/expat/all/conandata.yml:21]:
```suggestion

```

**Comment by SpaceIm** [comment]
> 
> 
> @SpaceIm I attempted the version check & raise, but that seemed to have issues that weren't obvious to me, so I changed over to a patch that modifies the CMake error to a warning.

Yes I thinks it's a better solution.

(You had an error because `ConanInvalidConfiguration` must be imported from `conans.errors`)

File: recipes/expat/all/conandata.yml (hunk 2 above)
Comment (datalogics-robb) [reply recipes/expat/all/conandata.yml]: I left the patch as it was generated by `git diff` and use `strip: 2` to remove `a/expat/` from the paths.
I find this is much easier to maintain than manually editing patches, as there are generally multiple files being patched.  For that reason, I'm a big fan of `strip: N` for patches.

Comment (SpaceIm) [reply recipes/expat/all/conandata.yml]: hooks don't allow `strip` keyword in conandata.yml, and I see that your patch still contains `a/expat`

File: recipes/expat/all/conandata.yml [hunk 3]
```diff
@@ -11,3 +11,11 @@ sources:
...
     sha256: bf42d1f52371d23684de36cc6d2f0f1acd02de264d1105bdc17792bbeb7e7ceb
     url: https://github.com/libexpat/libexpat/releases/download/R_2_2_10/expat-2.2.10.tar.gz
+  "2.3.0":
+    sha256: 89df123c62f2c2e2b235692d9fe76def6a9ab03dbe95835345bf412726eb1987
+    url: https://github.com/libexpat/libexpat/releases/download/R_2_3_0/expat-2.3.0.tar.gz
+patches:
+  "2.3.0":
+    - patch_file: "patches/relax-vs-restriction.patch"
+      base_path: "source_subfolder"
+
```
Comment (SpaceIm) [inline recipes/expat/all/conandata.yml:21]:
```suggestion
```

File: recipes/expat/all/conanfile.py [hunk 4]
```diff
@@ -35,6 +34,7 @@ def source(self):
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+
```
Comment (SpaceIm) [inline recipes/expat/all/conanfile.py:37]:
```suggestion
```

**Comment by datalogics-robb** [comment]
What went wrong?  I get a 403 when I try to see "Detals" on the CI run.

**Comment by SpaceIm** [comment]
Try to close/open the PR.

By the way, I've opened an issue about this hard error in CMakeLists: https://github.com/libexpat/libexpat/issues/454

**Comment by datalogics-robb** [comment]
reopening to get output from CI.

**Pull request closed without merging**